### PR TITLE
Reuse `sys::unix::cmath` on other platforms

### PR DESCRIPTION
**Pull request by CDirkx**
Reuse `sys::unix::cmath` on all non-`windows` platforms. 

`unix` is chosen as the canonical location instead of `unsupported` or `common` because `unsupported` doesn't make sense semantically and `common` is reserved for code that is supported on all platforms. Also `unix` is already the home of some non-`windows` code that is technically not exclusive to `unix` like `unix::path`.

### CODE REVIEW

File: library/std/src/sys/unix/cmath.rs [hunk 1]
```diff
@@ -1,32 +1,31 @@
 #![cfg(not(test))]
 
-use libc::{c_double, c_float};
```
Comment (CDirkx) [inline library/std/src/sys/unix/cmath.rs:3]: These functions are used to implement methods of `f32` and `f64`, so anything else then `c_double = f64` and `c_float = f32` would be wrong anyway. This makes the file reusable on platforms that don't have `libc` available.

**Comment by yaahc** [comment]
Everything here looks good to me but I'm not super familiar with the platform abstraction modules yet so I'd like to get a second set of eyes on this before approving.

cc @rust-lang/libs

**Comment by m-ou-se** [comment]
(Adding myself to the assignees so I won't forget to look at it tomorrow.)

File: library/std/src/sys/unix/cmath.rs [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
-
+// These symbols are all defined in `compiler-builtins`
```
Comment (m-ou-se) [inline library/std/src/sys/unix/cmath.rs:3]: This is only the case on some platforms, right? On most platforms, these are directly provided by `libm`.

**Comment by m-ou-se** [comment]
Looks good to me, other than the above comment.

File: library/std/src/sys/unix/cmath.rs (hunk 2 above)
Comment (CDirkx) [reply library/std/src/sys/unix/cmath.rs]: I changed the comment to "These symbols are all defined by `libm`, or by `compiler-builtins` on unsupported platforms."

**Comment by yaahc** [comment]
Looks like it's good to go then

@bors r+

**Pull request merged**